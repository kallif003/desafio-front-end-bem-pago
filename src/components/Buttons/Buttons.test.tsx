import * as React from "react"
import { render } from "@testing-library/react"
import userEvent from "@testing-library/user-event"

import { SubmitButton, PlusAndMinusButton } from "./index"

describe("Buttons components", () => {
	it("Renders Submit button properly", () => {
		const { container } = render(<SubmitButton>Enviar</SubmitButton>)
		expect(container.firstChild).toMatchSnapshot()
	})
	it("Renders Plus and minus button properly", () => {
		const { container } = render(<PlusAndMinusButton>+</PlusAndMinusButton>)
		expect(container.firstChild).toMatchSnapshot()
	})
	it("test Submit button ", () => {
		const { container } = render(<SubmitButton>Enviar</SubmitButton>)
		userEvent.click(container)
	})
	it("test Plus and minus button ", () => {
		const { container } = render(<PlusAndMinusButton>+</PlusAndMinusButton>)
		userEvent.click(container)
	})
})
