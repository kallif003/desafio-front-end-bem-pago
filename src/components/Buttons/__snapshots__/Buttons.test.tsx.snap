// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons components Renders Plus and minus button properly 1`] = `
<button
  class="sc-gsDKAQ dlHsdh"
>
  +
</button>
`;

exports[`Buttons components Renders PlusAndMinusButton properly 1`] = `
<button
  class="sc-gsDKAQ dlHsdh"
>
  Test
</button>
`;

exports[`Buttons components Renders Submit button properly 1`] = `
<button
  class="sc-bdvvtL kDPKLz"
>
  Enviar
</button>
`;

exports[`Buttons components Renders SubmitButton properly 1`] = `
<button
  class="sc-bdvvtL kDPKLz"
>
  Test
</button>
`;
